crates/vm/src/system/program/trace.rs

```diff
@@ -62,7 +62,9 @@ impl<F: PrimeField64> ProgramChip<F> {
         let common_trace = RowMajorMatrix::new_col(
             self.execution_frequencies
                 .into_iter()
-                .map(|x| F::from_canonical_usize(x))
+                .enumerate()
+                .filter(|(i, _)| self.program.get_instruction_and_debug_info(*i).is_some())
```

Comment:
can you zip `execution_frequences` with `program.instructions_and_debug_infos` instead? it feels slightly more canonical (and also prevent a `get` for every index)

Reply:
You can also use `filter_map` (or `flat_map`) instead of `filter` and `map`.